### ISSUE TITLE
Fill store/fetch data on route prefetch

### ISSUE THREAD

**Issue by Timkor** [nuxt/nuxt#4898]
### What problem does this feature solve?

Routes are now able to be prefetched similair to quicklink. The problem is that there still needs to be  data fetched for some routes which implement `asyncData` and/or `fetch`. This can be solved by enabling the page to prefetch data or fill the store before the route is being requested.

Prefetching can be done before the actual route transition. That means, that it's possible that the data is fetched while the route transition will never happen. Some Nuxt.js project might have implemented `fetch` and `asyncData` in a way that it requires the route transition to happen.

### What does the proposed changes look like?
There could either be a new method being introduced called `prefetch` on page components or there could be a flag which enables prefetching on `fetch` and/or `asyncData`. This way, this feature is backwardscompatible.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8571">#c8571</a>)</em></sub></div>

**Comment by Atinux** [comment]
I had the similar idea @Timkor when thinking of the prefetching feature 😄 

This feature is worth exploring, but only after the new `asyncData` that could be called everywhere (and not only in `pages/`).

**Comment by danielroe** [comment]
We are approaching the Nuxt 2 EOL date (June 30, 2024) - see [this article](https://nuxt.com/blog/nuxt2-eol) for more information. I'm closing this issue as it's marked as a _Nuxt 2 related enhancement_ and it's not a critical issue.

That doesn't mean it might not be relevant for Nuxt 3. If it is, please feel free to open a new issue (or just comment, and I can reopen it). 🙏

Thank you for your understanding and for your contribution to Nuxt! 🎉
